src/poliastro/coordinates.py

```diff
@@ -100,22 +313,16 @@ def body_centered_to_icrs(r, v, source_body, epoch=None, rotate_meridian=False):
 
     """
 
-    # All epochs must be in TDB
-    if not epoch:
-        epoch = time.Time('J2000', scale='tdb')
-
-    T = (epoch.tdb - J2000).to('day').value / 36525
-    d = (epoch.tdb - J2000).to('day').value
-
+    ra, dec, W = rot_elements_from_body(source_body, epoch)
     if rotate_meridian:
-        r = transform(r, -bodies_orientations[source_body.name]['W'], 'z', u.deg)
-        v = transform(v, -bodies_orientations[source_body.name]['W'], 'z', u.deg)
+        r = transform(r, -W.value, 'z', W.unit)
```

Comment:
Are you sure that `transform` does not work with `Quantity` objects? Last time I used it, it did. Perhaps you can simplify the calls here.